site/public/js/gradeable.js

```diff
@@ -373,6 +373,41 @@ function ajaxUploadGradeable() {
     return true;
 }
 
+// Uploads a gradeable via JSON POST request
+function ajaxDownloadGradeable($gradeable_id) {
```

Comment:
Could do something like this

```js
//downloads a gradeable's json config
function downloadGradeableJsonConfig
    fetch(buildCourseUrl([gradeable_id, 'download']))
        .then( response => response.json() )
        .then( json => {
            if (json['status'] === 'success') {
                const temporaryElement = document.createElement('a');
                const file = new Blob([JSON.stringify(json['data'], null, 4)]);
                temporaryElement.href = URL.createObjectURL(file);
                temporaryElement.download = `${json['data']['id']}.json`;
                temporaryElement.click();
            }
            else{
                displayErrorMessage(json['message']);
            }
        })
        .catch(err => {
            displayErrorMessage(err);
        });
}
```

Fetch is a newer API thats a little bit less terse then ajax calls, there's also the ability to do `await` and `async` to mark when a function is used async vs not so thats a bit of older nomenclature the codebase used before it was standardized where we put `ajax` in front of the function to indicate it would do something async or make a request. Now you probably don't need and its more important to give a good descriptive name, especially one in a js file because it gets included in the global scope and will reduce the chance of a name collision 

The `displayErrorMessage` is a function defined globally in `server.js` and would show the nice standard red error banners we have around the site.

createObjectURL will also handle the encoding for the tmp link so you don't have to deal with uri encoding yourself and setting the preamble for the url

based on: https://stackoverflow.com/questions/34156282/how-do-i-save-json-to-local-text-file